pgrx-pg-sys/src/submodules/elog.rs

```diff
@@ -96,195 +96,333 @@ impl From<i32> for PgLogLevel {
+        errcode = ERRCODE_SUCCESSFUL_COMPLETION
+        docs = ""
+        extra_docs = true
+        extra_stmts = {}
+    }
 }
 
-/// Log to Postgres' `notice` log level.
-///
-/// This macro accepts arguments like the [`println`] and [`format`] macros.
-/// See [`fmt`](std::fmt) for information about options.
-#[macro_export]
-macro_rules! notice {
-    ($($arg:tt)*) => (
-        {
-            extern crate alloc;
-            $crate::ereport!($crate::elog::PgLogLevel::NOTICE, $crate::errcodes::PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION, alloc::format!($($arg)*).as_str());
-        }
-    )
+define_elog! {
+    info = {
+        loglevel = INFO
+        errcode = ERRCODE_SUCCESSFUL_COMPLETION
+        docs = ""
+        extra_docs = false
+        extra_stmts = {}
+    }
 }
 
-/// Log to Postgres' `warning` log level.
-///
-/// This macro accepts arguments like the [`println`] and [`format`] macros.
-/// See [`fmt`](std::fmt) for information about options.
-#[macro_export]
-macro_rules! warning {
-    ($($arg:tt)*) => (
-        {
-            extern crate alloc;
-            $crate::ereport!($crate::elog::PgLogLevel::WARNING, $crate::errcodes::PgSqlErrorCode::ERRCODE_WARNING, alloc::format!($($arg)*).as_str());
-        }
-    )
+define_elog! {
+    notice = {
+        loglevel = NOTICE
+        errcode = ERRCODE_SUCCESSFUL_COMPLETION
+        docs = ""
+        extra_docs = false
+        extra_stmts = {}
+    }
 }
 
-/// Log to Postgres' `error` log level.  This will abort the current Postgres transaction.
-///
-/// This macro accepts arguments like the [`println`] and [`format`] macros.
-/// See [`fmt`](std::fmt) for information about options.
-#[macro_export]
-macro_rules! error {
-    ($($arg:tt)*) => (
-        {
-            extern crate alloc;
-            $crate::ereport!($crate::elog::PgLogLevel::ERROR, $crate::errcodes::PgSqlErrorCode::ERRCODE_INTERNAL_ERROR, alloc::format!($($arg)*).as_str());
-            unreachable!()
-        }
-    );
+define_elog! {
+    warning = {
+        loglevel = WARNING
+        errcode = ERRCODE_WARNING
+        docs = ""
+        extra_docs = false
+        extra_stmts = {}
+    }
 }
 
-/// Log to Postgres' `fatal` log level.  This will abort the current Postgres backend connection process.
-///
-/// This macro accepts arguments like the [`println`] and [`format`] macros.
-/// See [`fmt`](std::fmt) for information about options.
-#[allow(non_snake_case)]
-#[macro_export]
-macro_rules! FATAL {
-    ($($arg:tt)*) => (
-        {
-            extern crate alloc;
-            $crate::ereport!($crate::elog::PgLogLevel::FATAL, $crate::errcodes::PgSqlErrorCode::ERRCODE_INTERNAL_ERROR, alloc::format!($($arg)*).as_str());
-            unreachable!()
-        }
-    )
+define_elog! {
+    error = {
+        loglevel = ERROR
+        errcode = ERRCODE_INTERNAL_ERROR
+        docs = "This will abort the current Postgres transaction."
+        extra_docs = false
+        extra_stmts = {unreachable!()}
```

Comment:
We don't need this. This code is already non-returning and this doesn't emit any optimization hints or anything like that.